test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -928,6 +930,52 @@ UCS_TEST_P(test_pci_bw, get_pci_bw)
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_pci_bw, all, "all")
 
+class rlimit_setter {
+public:
+    rlimit_setter(int resource, rlim_t new_limit) :
+        m_ret(-1), m_resource(resource), m_limit(1024), m_orig({})
+    {
+        struct rlimit limit;
+
+        m_ret = getrlimit(m_resource, &m_orig);
+        if (m_ret != 0) {
+            UCS_TEST_MESSAGE << "could not get rlimit";
+            return;
+        }
+
+        limit.rlim_cur = new_limit > m_orig.rlim_max ? m_orig.rlim_max :
+                                                       new_limit;
+        limit.rlim_max = m_orig.rlim_max;
+
+        if (setrlimit(m_resource, &limit) == 0) {
+            m_limit = limit.rlim_cur;
+            UCS_TEST_MESSAGE << "set rlimit to " << m_limit;
+        } else {
+            m_limit = m_orig.rlim_cur;
+            UCS_TEST_MESSAGE << "could not set rlimit to " << limit.rlim_cur;
+        }
+    }
+
+    ~rlimit_setter()
+    {
+        if (m_ret == 0) {
+            UCS_TEST_MESSAGE << "restored rlimit";
+            (void)setrlimit(m_resource, &m_orig);
+        }
+    }
+
+    rlim_t limit() const
+    {
+        return m_limit;
+    }
+
+private:
+    int           m_ret;
```

Comment:
maybe instead we can compare if m_limit==m_orig.rlim_cur and if equal do not need to restore during the dtor. so domn't need to keep m_ret variable.

Reply:
fixed